acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -394,3 +394,112 @@ def have_all_rpc_responses?(targets, responses)
   end
   return true
 end
+
+# A suite of functions for creating a manifest that sleeps for a long period
+# and interacting with Puppet runs of that manifest.
+def create_sleep_manifest(master, manifest, seconds_to_sleep)
+    create_remote_file(master, manifest, <<-MODULEPP)
+node default {
+  case $::osfamily {
+    'windows': { exec { 'sleep': command => 'C:/Windows/system32/ping.exe 127.0.0.1 -n #{seconds_to_sleep}', returns => 1 } }
+    'Darwin':  { exec { 'sleep': command => '/bin/sleep #{seconds_to_sleep} || /usr/bin/true', } }
+    default:   { exec { 'sleep': command => '/bin/sleep #{seconds_to_sleep} || /bin/true', } }
+  }
+}
+MODULEPP
+    on(master, "chmod 644 #{manifest}")
+end
```

Comment:
in the puppet repo we have a `#create_sitepp`  we might want to use that here later, but it shouldn't hold up the merge.